### PR TITLE
feat: check for duplicate maintainer/members within team

### PR DESCRIPTION
**Pull request by erickzhao**
Ref https://github.com/electron/.permissions/pull/202

Some formatting changes got pulled in via Prettier when I saved (/shrug).

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -61,27 +61,46 @@ async function run() {
...
+        });
+
+        if (maintainerAndMember.length > 0) {
+          core.setFailed(
+            `GitHub team ${
+              team.name
+            } has one or more maintainers who are also duplicated as members: ${maintainerAndMember.join(
+              ', ',
+            )}`,
+          );
```
Comment (dsanders11) [inline index.js:93]:
```suggestion
          );
          return;
```

**Review by dsanders11 — APPROVED** [review]

File: index.js [hunk 2]
```diff
@@ -61,27 +61,47 @@ async function run() {
...
       if (!team.formation && !hasMaintainers(team)) {
         core.setFailed(
           `GitHub team ${team.name} does not have valid maintainer(s)`,
         );
         return;
       }
 
+      // Check that maintainers are not duplicated in the members field
+      if (hasMaintainers(team)) {
+        const maintainerAndMember = team.maintainers.filter((maintainer) => {
```
Comment (codebytere) [inline index.js:82]: perhaps a bit simpler here to use `Array.some()`?

```
if (team.maintainers.some(m => team.members.includes(m))) {
  core.setFailed(
    `GitHub team ${
      team.name
    } has one or more maintainers who are also duplicated as members`,
  );
  return;
}
```

**Review by codebytere — APPROVED** [review]
LGTM with possible improvement!

File: index.js (hunk 2 above)
Comment (erickzhao) [reply index.js:82]: 👀 I had that originally but the output is clearer this way because we collect the member names!

**Review by codebytere — APPROVED** [review]

**Pull request merged**